File: cpp/rna-transcription/rna_transcription.cpp
#include "rna_transcription.h"

#include <iostream>

namespace rna_transcription {
  char to_rna(const char c) {
    return complements.at(c);
  }

  std::string to_rna(const std::string seq) {
    std::string str = std::accumulate(seq.begin(), seq.end(), std::string(""),
        [](auto& cs, const auto& c) { 
        char co = to_rna(c);
        return cs.append(1UL, co); });
    return str;
  }

  std::string to_rna(const char* str) {
    return to_rna(std::string(str));
  }
}

File: erlang/rna-transcription/src/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).

-spec to_rna(string()) -> string().
to_rna([]) -> [];
to_rna([H|T]) -> [to_dna_char(H) | to_rna(T)].

-spec to_dna_char(string()) -> string().
to_dna_char($G) -> $C;
to_dna_char($C) -> $G;
to_dna_char($T) -> $A;
to_dna_char($A) -> $U;
to_dna_char(_) -> throw(error).
